test/Microsoft.IdentityModel.Xml.Tests/EnvelopedSignatureReaderTests.cs

```diff
@@ -213,8 +212,8 @@ public static TheoryData<EnvelopedSignatureTheoryData> ReadXmlElementsTheoryData
+                var samlpTokenKey = new X509SecurityKey(X509CertificateHelper.Load(Convert.FromBase64String("MIIGvzCCBKegAwIBAgICAZUwDQYJKoZIhvcNAQELBQAwgYYxCzAJBgNVBAYTAkNaMQ8wDQYDVQQHEwZQcmFndWUxGTAXBgNVBAoTEENaLk5JQywgei5zLnAuby4xMTAvBgNVBAMTKENaLk5JQyBTSEEyIFJvb3QgQ2VydGlmaWNhdGlvbiBBdXRob3JpdHkxGDAWBgkqhkiG9w0BCQEWCWNhQG5pYy5jejAeFw0xODAyMDYxMDM5MDJaFw0yMDAyMDYxMDM5MDJaMH0xCzAJBgNVBAYTAkNaMQ8wDQYDVQQHEwZQcmFndWUxDzANBgNVBAoTBkNaLk5JQzEwMC4GA1UEAxMnbW9qZWlkLnJlZ3Rlc3QubmljLmN6IHNpZGFzYW1sIG1ldGFkYXRhMRowGAYJKoZIhvcNAQkBFgtyb290QG5pYy5jejCCAiIwDQYJKoZIhvcNAQEBBQADggIPADCCAgoCggIBANUq8haz+ZNnLBWRLPLQKSyg7TlQDVyLxxMmxFhCT8OXArkjIsZvcwNNeMDB6aOhk0Zs97t+7lEgC2MuiY9GnoC0DV+TAUf7+MzlHuE0oCmR9jiJPsikBEsRDHpvHi9rZv4HK2TCsee/5dDi7tP7bMnRvDOAt7lR+KuLQNaXtDWrXC4bjeNdO/mcy3UeKy+dW2Diqz8YMbRrxM29wAweaUSJ6npU9KTnx0/dq/+IM4R1gO62t+6vjxqiryEFcvdb6lGHc6qC9TYuHaGZBfXiT2goK4NbOr9dfLuixQ8Jd3oN88Qqt7r5u20VLCB06BIQHBgzJTHaSsi5MT5ymtx8lTpkR6MHXGue//QTZPi5DVBonb1B+ilgWdG7jK5yTBA6BkQSbbFp4uHM3IWdExErV/FPeyN9T7Au0kf7Jp73m7gMjD6ytC9xSI082ELufkjmerLTB0SoNPEsfAzUgDQeJ4DhsWg9kiK1/nhakjENefVW3FA2rZYBsRZkZr/uGdg/XEnw34ooeh+sTsj4QF5nWeuGmq0nu08hSTLv6YYfGwJny0TNekmfNoNL7Ip1RoRenl2ayruqvMSEzh4z5D1m4hW6zwmsRj0X8FJOk9pOr0NbHVsr2RuefmKNntk2bXMYq8dO3xSRASmdVmgmoyoQGJrnh1E0SihxCFNgiHqyGQ1JAgMBAAGjggE9MIIBOTAJBgNVHRMEAjAAMBEGCWCGSAGG+EIBAQQEAwIEsDAdBgNVHQ4EFgQU0HAvIqGfYUW/akxHd9SNK2O4GwEwgbsGA1UdIwSBszCBsIAUzxJQXpMfXwleF48WV0F6n3lyhLihgYykgYkwgYYxCzAJBgNVBAYTAkNaMQ8wDQYDVQQHEwZQcmFndWUxGTAXBgNVBAoTEENaLk5JQywgei5zLnAuby4xMTAvBgNVBAMTKENaLk5JQyBTSEEyIFJvb3QgQ2VydGlmaWNhdGlvbiBBdXRob3JpdHkxGDAWBgkqhkiG9w0BCQEWCWNhQG5pYy5jeoIJAJxtRGsvNinfMBQGA1UdEgQNMAuBCWNhQG5pYy5jejAWBgNVHREEDzANgQtyb290QG5pYy5jejAOBgNVHQ8BAf8EBAMCBaAwDQYJKoZIhvcNAQELBQADggIBAIytUqXh7AU/OunkSpUTaEY5Ze2sdV76JPwYVzNn2O6hCyzGvPXARP0IchUv8Vy+eCFLctUxvCbUS6aC+ObKgGXq4MxoSV8lijMLEW9crCpFDoLLd3LQw0GjVk2mCE7XTuaT0choPYlZmjv+wF2ZKm5/B+Qjek2j7SkY1yn7hxgJdd5ljHE6wmDXLJ8gHuVBNwvc5iHDjHYh7jL5c5jCDBcr1fFCsIARU05RAkfpWurl8GKY8t6IPm7iopOLjru3Gl45ZBdVrAPMQ8Fz1M9VElUJ6ngeKkXHkSwGhCBG3X0MYsltND6mZSkJqN0nOs+cJ6HHO8IZW1f2pH3aCTUSDYWoaZbK74NC6d61sr5Rth4foLQnMzCS5RaXuANMvyZW3Ol5ScvLl/KRZM4f4CB6rmYinyHfXIoPF+uCjavyOYnW1RDBASg0Ld/WUJlWb75m5GNkRELIc4c5FU54ysMW9o5wnGpQvtXNdCBJK8tAyZO9Wf2hjZeOZgJ6r1IngfSeSFu7EOFqWnwVOF+3juwWOLCwxrKcURAEngwH01ydwU3oG/rN+7JtdS3IwfaBt9sfDiLQ60qec/6PQc643UztE6oToHLRXsidwrObwyAKLSFJoh/uxWT85JgAoekq5zBen94HfELfMEc9tex6Qlf1tLDs7OWD6Mlw6j9aAcw/4Nfh")));
```

Comment:
Had to add a `X509CertificateHelper` to this project as adding an internals visible to for `Microsoft.IdentityModel.Tokens.Xml.Tests` to the Wilson.Tokens package assembly info didn't work like it did for Wilson.Protocols.WsFederation, for example. The only thing I can think of is because the namespace of this file doesn't match the package/.csproj, which is `Microsoft.IdentityModel.Xml.Tests` and not `Microsoft.IdentityModel.Tokens.Xml.Tests`.

Reply:
![image](https://github.com/user-attachments/assets/7f9edbb2-9bb7-4a9b-95d2-c7526288d071)
Despite adding to Wilson.Tokens assembly info, adding Wilson.Tokens package ref to .csproj, using Wilson.Tokens at the top of this file..

Reply:
Not sure I'm following, so the internalsVisibleTo did not work?

Reply:
Did not work, some of the tests in this project have a different namespace than the project namespace.

Reply:
I'll be adding a public helper to the tests class to avoid sharing internals and this issue will be fixed